ramalama/http_client.py

```diff
@@ -66,11 +67,14 @@ def perform_download(self, file, progress):
                 accumulated_size += size
                 if time.time() - last_update_time >= 0.1:
                     self.update_progress(accumulated_size)
+                    do_final_progress = True
                     accumulated_size = 0
                     last_update_time = time.time()
 
-        if accumulated_size:
-            self.update_progress(accumulated_size)
+        if do_final_progress:
+            if accumulated_size:
```

Comment:
This makes no sense, to add the option.

Why not do 

if accumulated_size > 0:

You reset the accumalated_size option right after the do_final_progress.

Reply:
Ah it's the "\033[K" outside the if block that's the main fix...

I dislike python's indenting as '{' '}' sometimes, makes things like this not so obvious :)

Reply:
Wait sorry this does make sense... It's just not clear, I'll do it a different way that's more clear

Reply:
So you can break out with `if not data:` when you have accumalated size and do not want to update_progress?